File: Programmers/nolv/12943.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int num) {
    int answer = 0;
    long ltemp = (long)num;
    while (answer < 500 && ltemp != 1) {
        if (ltemp % 2 == 0) {
            ltemp /= 2;
        }
        else {
            ltemp = ltemp * 3 + 1;
        }
        answer++;
    }

    if (ltemp != 1)
        answer = -1;

    return answer;
}

File: Python/12943.py
def solution(num):
    answer = 0
    while True:
        if answer > 500:
            return -1
        if num%2 == 0:
            num = num//2
        elif num == 1:
            break
        else:
            num = num * 3 + 1
        answer = answer + 1

    return answer